wgpu-core/src/device/queue.rs

```diff
@@ -143,12 +143,56 @@ crate::impl_storage_item!(Queue);
 impl Drop for Queue {
     fn drop(&mut self) {
         resource_log!("Drop {}", self.error_ident());
+
+        let last_successful_submission_index = self
+            .device
+            .last_successful_submission_index
+            .load(Ordering::Acquire);
+
+        let fence = self.device.fence.read();
+        let wait_res = unsafe {
```

Comment:
nitpick: Missing a `SAFETY` comment here.

Reply:
I don't know what the safety requirements of calling this function are... I'd have to check and document all the backends for this. All wgpu-hal functions are marked as unsafe without a safety contract. We should document the safety contract at some point but I don't think this is the time for that.